### ISSUE TITLE
Quests randomly resetting

### ISSUE THREAD

**Issue by Sjilli** [FlyingPikachu/Quests#188]
(__Required__) What is your Quests version / build number (do _not_ say "latest")?
Dev build 196
(__Required__) Please describe the problem and how to replicate it. Any console errors?
No console errors, players randomly lose their progress.. This hasnt happened on a small quest only the bigger ones such as killing 500 zombies. It doesnt always occur it's random.
(__Optional__) Can you provide an example quest from quests.yml _using Pastebin.com_?
https://pastebin.com/PzM9ndje

**Comment by Sjilli** [comment]
Can someone help me with this..?

**Comment by FlyingPikachu** [comment]
@Sjilli After a little experimenting, I believe you may be dealing with a bug that was fixed recently. Please give the latest dev build a try ([202 at time of writing](http://ci.ac3-servers.eu/job/Quests/202/)) and let us know if the issue persists (if so, check for new errors).

**Comment by Sjilli** [comment]
Alright, if I replace my current quests.jar with this one, will it replace my quests and I'll have to re-add them?
Thanks in advance.

**Comment by FlyingPikachu** [comment]
@Sjilli No, the build I linked will not replace any Quests you have made, so long as you do _not_ delete quests.yml

**Comment by Sjilli** [comment]
Brilliant, thank you very much!

**Comment by Sjilli** [comment]
This issue still persists.. Please fix it ASAP.. seriously.

**Comment by FlyingPikachu** [comment]
@Sjilli Easier said than done, as I cannot recreate this in my test environment. Could you provide a server log for the time-frame during which this occurred?

**Comment by Sjilli** [comment]
As I said, it's completely random and there are no error logs..

It's really confusing.

**Comment by Sjilli** [comment]
I accidentally clicked close.. whoops?

**Comment by FlyingPikachu** [comment]
@Sjilli Server logs include startup messages and plugin outputs; not just errors. Unless you're able to provide one, I'm afraid there's little more for me to examine/try.

**Comment by Sjilli** [comment]
https://pastebin.com/twFifgnG Here is the startup log, I've seen some errors saying the quests dont exist, however they do, and these aren't the quests being effected anyway.

**Comment by FlyingPikachu** [comment]
@Sjilli Nonetheless, those errors are now the best lead we have. Seeing as you've already posted the "Zombie Terminator" quest, would you mind posting the "Zombie Executioner" quest as well?

Errors for reference:

> [22:53:31] [Server thread/ERROR]: [Quests] Zombie Terminator inside quests: Requirement for Quest Zombie Executioner is not a valid Quest name!
[22:53:31] [Server thread/ERROR]: [Quests] Make sure you are using the Quest name: value, and not the block name.
[22:53:31] [Server thread/ERROR]: [Quests] The Infection I inside quests: Requirement for Quest The Infection II is not a valid Quest name!
[22:53:31] [Server thread/ERROR]: [Quests] Make sure you are using the Quest name: value, and not the block name.
[22:53:31] [Server thread/ERROR]: [Quests] Zombie Horde inside quests: Requirement for Quest Baby Zombies is not a valid Quest name!
[22:53:31] [Server thread/ERROR]: [Quests] Make sure you are using the Quest name: value, and not the block name.

**Comment by Sjilli** [comment]
Sure, https://pastebin.com/4662Cvjz is the link to the Zombie Executioner quest.

Hope we can resolve this :)

**Comment by Sefreneth** [comment]
I had the same bug now, it happened to one player only though. Some daily quests got reset after restart, even thou it was not 24 hours yet

**Comment by FlyingPikachu** [comment]
Since I first tackled this nearly a year ago, I've been unable to recreate this in test environment across multiple versions of Quests, Spigot, and Java. Today, I gave it one last go, but could not replicate the problem. It's entirely possible this issue has been resolved as the result of another ticket, or it may be OS dependent (I've been using Windows 7).

It's time this issue be closed. It may be reopened if new evidence/leads are brought forward. For future reference, the quests.yml file I used for testing is linked below.

https://pastebin.com/Hc5MW8ZY
